src/freenet/clients/http/templates/first-time-wizard.html

```diff
@@ -0,0 +1,115 @@
+                        <div class="error">{{ errors.uploadLimitError }}</div>
+                    {% endif %}
+                    <input id="downLimit" name="downLimit" type="number" min="0" value="{{ downloadLimit }}">
+                    {{ l10n("bandwidthDownLimitBaseUnit") }}.
+                    <input id="upLimit" name="upLimit" type="number" min="0" value="{{ uploadLimit }}">
+                    {{ l10n("bandwidthUpLimitBaseUnit") }}.
+                <p>
+                <h4>{{ l10n("bandwidthCommonInternetConnectionSpeeds") }}</h4>
+                <table>
+                    <tr>
+                        <th>{{ l10n("bandwidthCommonInternetConnectionSpeedsConnectionType") }}</th>
+                        <th>{{ l10n("bandwidthCommonInternetConnectionSpeedsDownloadLimit") }}</th>
+                        <th>{{ l10n("bandwidthCommonInternetConnectionSpeedsUploadLimit") }}</th>
+                    </tr>
+                    <tr>
+                        <td>{{ l10n("bandwidthCommonInternetConnectionSpeedsDetected") }}</td>
+                        <td>{{ downloadLimitDetected }} {{ l10n("bandwidthCommonInternetConnectionSpeedsKibps") }}</td>
+                        <td>{{ uploadLimitDetected }} {{ l10n("bandwidthCommonInternetConnectionSpeedsKibps") }}</td>
+                    </tr>
+                    <!-- TODO: This will be a template loop over a container of common speeds provided by the node. .type, .down, .up per row. -->
+                    <tr>
+                        <td>{{ l10n("bandwidthCommonInternetConnectionSpeeds4Mbit") }}</td>
+                        <td>256 {{ l10n("bandwidthCommonInternetConnectionSpeedsKibps") }}
+                            (2 {{ l10n("bandwidthCommonInternetConnectionSpeedsMibs") }})
+                        </td>
+                        <td>16 {{ l10n("bandwidthCommonInternetConnectionSpeedsKibps") }}</td>
+                    </tr>
+                    <tr>
+                        <td>{{ l10n("bandwidthCommonInternetConnectionSpeeds6Mbit") }}</td>
+                        <td>384 {{ l10n("bandwidthCommonInternetConnectionSpeedsKibps") }}
+                            (3 {{ l10n("bandwidthCommonInternetConnectionSpeedsMibs") }})
+                        </td>
+                        <td>16 {{ l10n("bandwidthCommonInternetConnectionSpeedsKibps") }}</td>
+                    </tr>
+                </table>
+                </p>
+                </p>
+            </div>
+        </fieldset>
+        <fieldset>
+            <legend>{{ l10n("storage") }}</legend>
+            <p>
+                {% if errors.storageLimitError is not null %}
+                    <div class="error">{{ errors.storageLimitError }}</div>
+                {% endif %}
+                Give
+                <!-- TODO: Populated by the node with some percentage of free space. -->
+                <!-- TODO: Possibly attach this number with a slider of percentage of free space. -->
```

Comment:
Agreed that this would be nice, but definitely not necessary for an initial implementation.

Reply:
I agree — I’ll leave in the TODO as a reminder.